great_expectations/core/factory/suite_factory.py

```diff
@@ -103,9 +104,24 @@ def get(self, name: str) -> ExpectationSuite:
     def all(self) -> Iterable[ExpectationSuite]:
         """Get all ExpectationSuites."""
         dicts = self._store.get_all()
-        # Even though we deserialize here, all the validation has already
-        # been done via marshmallow in the get_all() call
-        return [self._deserialize(suite_dict) for suite_dict in dicts]
+        # Marshmallow validation was done in the previous get_all() call for
+        # suites but we can still die here because pydantic validation happens
+        # on the expectations inside the suites here.
+        # TODO: deserialization should not live in the factory and should
+        # TODO: live in the store like in other domain objects. That will
+        # TODO: allow us delete this error handling here.
+        deserializable_suites: list[ExpectationSuite] = []
+        bad_dicts: list[Any] = []
+        for suite_dict in dicts:
+            try:
+                deserializable_suites.append(self._deserialize(suite_dict))
+            except PydanticValidationError as e:
+                bad_dicts.append(suite_dict)
+                self._store.submit_all_deserialization_event(e)
```

Comment:
Should the analytics event method live on the store? Just thinking out loud I guess I don't have any major qualms

Reply:
For now, I think so because that is where deserialization happens for checkpoints and validations. For suites, it half lives there (for Marshmallow suite level validation) and half here (for expectations validation via pydantic). Once we move suites to pydantic, deserialization should all live in the store. This leaks because we're in this 1/2 state that already leaks deserialization into the factory.